### PR TITLE
Styling Improvements

### PR DESCRIPTION
**Pull request by rssoc**
Minor improvements on the way the code is read.

Loops have been transformed into their map equivalent, and indention space has been saved.

### CODE REVIEW

File: src/utils.lisp [hunk 1]
```diff
@@ -35,29 +35,27 @@
...
+  (string-trim '(#\Space #\Newline #\Backspace #\Tab
+                 #\Linefeed #\Page #\Return #\Rubout)
                str))
 
 (defun instance-has-backgrounds-p ()
-  (> (length  (loop for (key . value)
-                      in *boards*
-                    when (cassoc :background value)
-                      collect (cassoc :background value)))
-     0))
```
Comment (rssoc) [inline src/utils.lisp:54]: In Common Lisp, anything that is not the empty list NIL or '() is considered true.
No need to check length against 0.

**Comment by rssoc** [comment]
I was not able to fully test the changes, so please do some tests.

File: src/views.lisp [hunk 2]
```diff
@@ -71,7 +71,9 @@
                                                 (color-green rgb)
                                                 4)))
                        6)))
-            (list (round (* 360 h)) (round (* s 100)) (round (* 100 average))))))))
+            (list (round (* 360 h))
+                  (round (* s 100))
+                  (round (* 100 average))))))))
```
Comment (rssoc) [inline src/views.lisp:76]: This makes it easier to see the components of the list.

File: src/utils.lisp [hunk 3]
```diff
@@ -99,12 +99,14 @@
...
 ;; Request handling
 (defun api-success ()
-  (encode-json-alist-to-string (pairlis '(status) '("Success"))))
+  (encode-json-alist-to-string
+   (pairlis '(status) '("Success"))))
 
 (defun api-fail-with-message (msg code)
   (setf (return-code*) code)
-  (encode-json-alist-to-string (pairlis '(message status)
-                                        (list msg "Error"))))
```
Comment (rssoc) [inline src/utils.lisp:107]: Try to avoid going too far to the right like this. Use some newlines.

File: src/views.lisp [hunk 4]
```diff
@@ -201,14 +207,11 @@
...
-                      :prev-page (when (> page
-                                          0)
-                                   (- page 1))
-                      :next-page (when (or (> (length text-posts)
-                                              20)
-                                           (> (length file-posts)
-                                              20))
-                                   (+ page 1)))))
+                      :prev-page (when (> page 0)
+                                   (1- page))
```
Comment (rssoc) [inline src/views.lisp:211]: 1- and 1+ procedures are nice short hands.

File: src/views.lisp [hunk 5]
```diff
@@ -154,12 +161,11 @@
                                          (alist-keys *boards*))
                           :board-data (cassoc board *boards* :test #'string=)
                           :csrf-token (session-csrf-token)
-                          :next-page (when (or
-                                            file-posts-next-page-p
-                                            text-posts-next-page-p)
-                                       (+ page 1))
```
Comment (rssoc) [inline src/views.lisp:160]: Lispers read by indentation, so this reads a bit awkwardly

File: src/views.lisp [hunk 6]
```diff
@@ -135,12 +142,12 @@
          (file-posts (select-posts (+ 1 *file-display-count*) (* page *file-display-count*) :type "file" :board board))
          (text-posts-next-page-p (> (length text-posts) *text-display-count*))
          (file-posts-next-page-p (> (length file-posts) *file-display-count*))
-         (stylized-text-posts (map 'list #'stylize-text-post (if text-posts-next-page-p
```
Comment (rssoc) [inline src/views.lisp:138]: If you know your input is a list, you can just use mapcar. Map is more for generic operations against things like arrays and strings.

File: src/views.lisp [hunk 7]
```diff
@@ -102,17 +104,21 @@
...
+               (if (and corruptions-for-character
+                        (eql (random 2)
+                             1))
+                   (nth (random (length corruptions-for-character))
+                        corruptions-for-character)
+                   char)))
          text)))
 
 (defun stylize-text-post (post)
-  (acons :data (corrupt (cassoc :data post)) (acons :style (text-style (cassoc :board post)) post)))
```
Comment (rssoc) [inline src/views.lisp:115]: Avoid long one liners like this, use some newlines and indention to add clarity to the structure.

File: src/utils.lisp [hunk 8]
```diff
@@ -67,14 +65,15 @@
...
 
 (defun form-encoded-default-preferences ()
-  (url-encode-params (reduce #'(lambda (acc pref)
-                                 (acons (string-downcase (car pref))
-                                        (if  (cassoc :default (cdr pref) :test #'string=)
-                                             "on"
-                                             "off")
-                                        acc))
-                             *web-user-preferences*
-                             :initial-value '())))
```
Comment (rssoc) [inline src/utils.lisp:77]: Use a newline and save some indentation space.

File: src/routes.lisp [hunk 9]
```diff
@@ -42,24 +42,22 @@
...
             (let* ((posts (select-posts 30 (* page 30) :board board))
-                   (sorted-posts (sort posts
-                                       #'sort-posts-by-id)))
+                   (sorted-posts (sort posts #'sort-posts-by-id)))
               (with-output-to-string (s)
                 (with-rss2 (s :encoding "utf-8")
                   (rss-channel-header "nmebious" *web-url*
                                       :description "monitoring the wired")
                   (dolist (item sorted-posts)
-                    (let* ((data (cassoc :data item))
```
Comment (rssoc) [inline src/routes.lisp:59]: let* is used for when you want to reference the previous let bindings in your let (as the regular let cannot do this).
example: (hate how github doesn't preserve the indentation)
`
(let* ((a 5)
         (b (* a 2)))
  b)
`
let* is actually just a macro which expands to.
`
(let ((a 5))
  (let ((b (* a 2)))
     b))
`
and doing all this expansion is unnecessary when you don't need it

File: src/views.lisp [hunk 10]
```diff
@@ -178,7 +184,7 @@
 
 (defun render-preferences-page ()
   (let* ((user-prefs (parse-user-preferences))
-         (render-prefs (map 'list
+         (render-prefs (mapcar
                             #'(lambda (pref)
                                 (let* ((keyword-pref (car pref)))
                                   (car (acons keyword-pref (acons :current
```
Comment (rssoc) [inline src/views.lisp:190]: (Github is not letting me select exactly what I want)
Side-note: I did not fix it, but your lambda definition here goes too far too the right. Generally that's sign that some abstractions are hiding.
Same goes for procedures which are too long and condensed, generally it's a sign that too many ideas are being put under 1 name.

**Comment by ad044** [comment]
thanks so much for this, really appreciate it

**Pull request merged**